src/tests/comments.test.js

```diff
@@ -0,0 +1,123 @@
+    expect(error).to.equal('The are no comments on this request');
+  });
+
+  it('should create a comment with valid data', async () => {
+    const { body: { data: { comment, id } }, status } = await chai.request(app)
+      .post(`${URL_PREFIX}/${requestId}/comments`)
+      .set('token', authToken)
+      .send(commentBody);
+
+    commentId = id;
+    expect(status).to.equal(201);
+    expect(comment).to.equal('this is a coment');
+  });
+
+  it('should return an error  with invalid data', async () => {
+    const { body: { error: [first] }, status } = await chai.request(app)
+      .post(`${URL_PREFIX}/${requestId}/comments`)
+      .set('token', authToken)
+      .send('commentBody');
+
+    expect(status).to.equal(400);
+    expect(first).to.equal('Comment is required');
+  });
+
+  it('should return an error message for invalid request id', async () => {
+    const { body: { error }, status } = await chai.request(app)
+      .post(`${URL_PREFIX}/requestId/comments`)
+      .set('token', authToken)
+      .send(commentBody);
+
+    expect(status).to.equal(400);
+    expect(error).to.equal('invalid input syntax for integer: "requestId"');
+  });
+
+  it('should get requests comments', async () => {
+    const { body: { data: [first] }, status } = await chai.request(app)
+      .get(`${URL_PREFIX}/${requestId}/comments`)
+      .set('token', authToken)
+      .send();
+    const { comment } = first;
+
+    expect(status).to.equal(200);
+    expect(comment).to.equal('this is a coment');
+  });
+
+  it('should throw an erro for invalid request id', async () => {
+    const { body: { error }, status } = await chai.request(app)
+      .get(`${URL_PREFIX}/requestId/comments`)
+      .set('token', authToken)
+      .send();
+
+    expect(status).to.equal(400);
+    expect(error).to.equal('invalid input syntax for integer: "requestId"');
+  });
+
+  it('should update the deleted field', async () => {
+    const { body: { data }, status } = await chai.request(app)
+      .delete(`${URL_PREFIX}/${requestId}/comments/${commentId}`)
+      .set('token', authToken)
+      .send();
+
+    expect(status).to.equal(200);
+    expect(data).to.equal('One field updated');
+  });
+
+  it('should return an error when attempting to delete with an invalid comment id', async () => {
+    const { body: { error }, status } = await chai.request(app)
+      .delete(`${URL_PREFIX}/requestId/comments/commentId`)
+      .set('token', authToken)
+      .send();
+
+    expect(status).to.equal(400);
+    expect(error).to.equal('invalid input syntax for integer: "commentId"');
+  });
+
+  it('should return an error when attempting to delete comment that does not exist', async () => {
+    const { body: { error }, status } = await chai.request(app)
+      .delete(`${URL_PREFIX}/${requestId}/comments/666`)
+      .set('token', authToken)
+      .send();
+
+    expect(status).to.equal(400);
+    expect(error).to.equal('Comment does not exist');
```

Comment:
What happens when a user that is not the owner of a comment try to delete someone else comment? I don't think you implemented that

Reply:
I use the users id (which I get from the token verifying middleware)as a foreign key on the comments table  so a user with a different id cannot delete a comment